src/Core/GraphQl/Type/FieldsBuilder.php

```diff
@@ -209,30 +217,35 @@ public function getResourceObjectTypeFields(?string $resourceClass, ResourceMeta
             return $fields;
         }
 
-        if (!$input || 'create' !== $mutationName) {
+        if (!$input || 'create' !== $operationName) {
             $fields['id'] = $idField;
         }
 
         ++$depth; // increment the depth for the call to getResourceFieldConfiguration.
 
         if (null !== $resourceClass) {
             foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
-                $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $property, ['graphql_operation_name' => $subscriptionName ?? $mutationName ?? $queryName]);
+                $context = [
+                    'normalization_groups' => $operation->getNormalizationContext()['groups'] ?? null,
+                    'denormalization_groups' => $operation->getDenormalizationContext()['groups'] ?? null,
```

Comment:
The groups were not used before, why this change?

Reply:
This is a complex issue, I want ResourceMetadataCollectionFactory to depend on PropertyMetadataFactory so that we can compute identifiers. The SerializerPropertyMetadataFactory also relies on ResourceMetadataFactories to compute groups which leads to a recursive dependency. I'm still trying to look for a solution, getting the groups before calling the SerializerPropertyMetadataFactory seems to be the less complex thing to do.

Reply:
I think it can create issues in nested type construction, but I'm not really sure. It needs to be checked.